### PR TITLE
Fix the grading link for exam programming exercises

### LINKED ISSUE #3693: Breadcrumb Programming Exercises in Exams
### Describe the bug

In Exam Management, for Exercise groups and there Configure Grading the breadcrumb path is replaced by Exercise which is misleading.__

### To Reproduce

1. Create an exam with programming exercise 
2. Edit the grading of the exercise 
3. Look at the breadcrumbs (which point to exercises instead of exams)

### Expected behavior

The breadcrumbs shall include the exam not the exercises

### Screenshots

_No response_

### What browsers are you seeing the problem on?

Firefox

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by FrankeLukas**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.

### Motivation and Context
Closes #3693.

### Description

When clicking on "Grading" of an exam programming exercise you now get redirected to a grading page inside the exam, so that you can still use breadcrumbs properly.

### Steps for Testing

1. Create or find an exam with a programming exercise
2. Open the grading page and make sure it works as for course exercises
3. Make sure the breadcrumbs are correct and that you can navigate back to exercise groups or exams for example

Example testcase: https://artemistest2.ase.in.tum.de/course-management/1915/exams/103/exercise-groups

### CODE REVIEW

**Review by MDK5 — APPROVED** [review]
Tested on ts2, worked correctly

Code changes also look good to me

File: src/main/webapp/app/exam/manage/exam-management.route.ts [hunk 1]
```diff
@@ -588,6 +589,15 @@ export const examManagementRoute: Routes = [
         },
         canActivate: [UserRouteAccessService],
     },
+    {
+        path: ':examId/exercise-groups/:exerciseGroupId/programming-exercises/:exerciseId/grading/:tab',
+        component: ProgrammingExerciseConfigureGradingComponent,
+        data: {
+            authorities: [Authority.EDITOR, Authority.INSTRUCTOR, Authority.ADMIN],
```
Comment (imstreetspirit) [inline src/main/webapp/app/exam/manage/exam-management.route.ts:596]: Grading button is enabled for Instructor and higher roles.(I just look at the template file.) I think we should organize authorizations according to that.

**Comment by imstreetspirit** [comment]
tested on ts2, works as described. I just added small inline comment.

File: src/main/webapp/app/exam/manage/exam-management.route.ts (hunk 1 above)
Comment (FuchsDominik) [reply src/main/webapp/app/exam/manage/exam-management.route.ts:596]: On a second thought, as editors as there to create exercises, we should probably also allow those to access the grading page, or not?

Comment (FrankeLukas) [reply src/main/webapp/app/exam/manage/exam-management.route.ts:596]: I copied those from the other route, apparently the button in the template should use isAtLeastEditor.

Comment (FrankeLukas) [reply src/main/webapp/app/exam/manage/exam-management.route.ts:596]: I checked back with Stephan and grading should be accessible for editors, so I fixed the auth check in the template in https://github.com/ls1intum/Artemis/pull/3748/commits/6cfcfea477e34787acf7c8ac4b07700a14e5c409.

**Review by NicolasUllmann — DISMISSED** [review]
Tested on TS1. Everything worked as expected.

**Review by willergeorg — APPROVED** [review]
Tested on TS2. Works for me.

**Review by imstreetspirit — DISMISSED** [review]

**Comment by krusche** [comment]
@FrankeLukas could you resolve the merge conflicts?

**Comment by FrankeLukas** [comment]
Resolved the conflicts.

**Pull request merged**